src/mqtt.js

```diff
@@ -91,6 +91,13 @@ export class ARENAMqtt {
                 console.warn('Malformed message (no data field):', JSON.stringify(message));
                 return;
             }
+            // check topic
+            if (message) {
+                if (!message.destinationName.endsWith(theMessage.data.source)) {
```

Comment:
Foe these 3 checks, I'd also make it include the slash so objects cannot be prefixed like:
```js
if (!message.destinationName.endsWith(`/${theMessage.data.source}`)) {
```